addons/website_sale_picking/models/payment_acquirer.py

```diff
@@ -0,0 +1,28 @@
+from odoo import fields, models
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    is_onsite_acquirer = fields.Boolean('Is Onsite Acquirer',
+                                        default=False)
+
+    def _get_compatible_acquirers(self, *args, sale_order_id=None, website_id=None, **kwargs):
+        compatible_acquirers = super()._get_compatible_acquirers(*args, website_id=website_id, **kwargs)
+        onsite_carriers = self.env['delivery.carrier'].search(
+                ['&', ('website_published', '=', True), ('delivery_type', '=', 'onsite')]
+            ).exists()
```

Comment:
`.exists` right after a search is meaningless as it will always query the db to fetch the ids.

```suggestion
            )
```